cmd/soroban-cli/src/commands/contract/bindings/ts.rs

```diff
@@ -0,0 +1,50 @@
+use std::{fmt::Debug, path::PathBuf};
+
+use clap::{command, Parser};
+use soroban_spec::gen::ts;
+
+use crate::wasm;
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    wasm: wasm::Args,
+
+    /// where to place generated project
+    #[arg(long)]
+    root_dir: PathBuf,
+
+    #[arg(long)]
+    contract_name: String,
+
+    #[arg(long, alias = "id")]
+    contract_id: String,
```

Comment:
Can we also allow specifying `rpc-url` and `network-passphrase` in the no-network version? These are networked params, so I'm not sure it does. But it also feels weird to [hardcode them](https://github.com/stellar/rs-soroban-sdk/pull/871/files#diff-3cb416aff49f093298d5ed678a7d07d70b591905101ace4c5af2a80c0da2bd07R17-R29) in the generated template, even if they're overridable with environment variables.

Maybe this doesn't matter, since we'll have the networked version soon enough.

Reply:
You can mark this comment as resolved